### ISSUE TITLE
`trivially_copy_pass_by_ref` makes suggestion that does not compile

### ISSUE THREAD

**Issue by awused** [rust-lang/rust-clippy#7518]
I tried this code:

```rust
#[derive(Copy, Clone)]
struct Test {
    a: u32,
}

impl Test {
    const fn do_something(self, other: &Self) -> bool {
        self.a == other.a
    }
}
```

Clippy's suggestion does not compile:

```rust
    const fn do_something(self, other: self) -> bool {
```

Newest stable version.

**Comment by mikerite** [comment]
[Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=83047eb94d8750ed998599846295e591)
```
  Checking playground v0.0.1 (/playground)
warning: this argument (4 byte) is passed by reference, but would be more efficient if passed by value (limit: 8 byte)
  --> src/lib.rs:10:40
   |
10 |     const fn do_something(self, other: &Self) -> bool {
   |                                        ^^^^^ help: consider passing by value instead: `self`
   |
```

The suggestion should be to use `Self` not `self`.
